tools/git/hooks/src/pre_commit.py

```diff
@@ -5,25 +5,45 @@
 import subprocess
 import sys
 import tempfile
+import argparse
 
 def main(): # pragma: no cover
-    files = sys.argv[1:]
-    failed_files = commit_is_ready(files)
-    if failed_files:
-        print("\nYou may run the following command to repeat the check:")
-        print("Hint: You may need to be at the repository's root directory.")
-        print("\n" + " \\\n".join([sys.argv[0]] + failed_files) + "\n")
-        print('Aborting Commit.')
-        sys.exit(1)
+    args = parse_args()
+
+    if args['uncrustify_all_files']:
+        file_names = get_all_files()
+        files_to_check = filter_checkable_files(file_names)
+        check_uncrustify(files_to_check, uncrustify=True)
+    elif args['uncrustify']:
+        file_names = get_modified_files()
+        files_to_check = filter_checkable_files(file_names)
+        check_uncrustify(files_to_check, uncrustify=True)
+    else:
+        failed_files = commit_is_ready()
```

Comment:
with the positional argument (see below), pass `args['files']` to `commit_is_readd()`.